rocky/reports/views/base.py

```diff
@@ -16,23 +16,30 @@
 from reports.forms import OOITypeMultiCheckboxForReportForm
 from reports.report_types.definitions import Report, ReportType
 from reports.report_types.helpers import get_report_by_id
-from rocky.views.mixins import OctopoesView
+from rocky.views.mixins import OOIList
+from rocky.views.ooi_view import OOIFilterView
+
+REPORTS_PRE_SELECTION = {
+    "clearance_level": ["2", "3", "4"],
+    "clearance_type": "declared",
+}
+
+
+def get_selection(request: HttpRequest, pre_selection: Optional[Dict[str, Union[str, List[str]]]] = None) -> str:
+    if pre_selection is not None:
+        return "?" + urlencode(pre_selection, True)
+    return "?" + urlencode(request.GET, True)
```

Comment:
It's not really wrong or blocking in that the code is working as far as I can see, but I don't really understand the purpose of this function.

How it is implemented it, it will either always return `"?" + urlencode(pre_selection, True)` or `"?" + urlencode(request.GET, True)` depending on whether tWhy not just write `"?" + urlencode(REPORTS_PRE_SELECTION, True)`  or  `"?" + urlencode(request.GET, True)` instead of calling this function? REPORTS_PRE_SELECTION is also only used in one place so it would also make sense to keep that as class variable as it was before.

Reply:
Yes we can indeed use your suggestion. i made this function to use for general purposes, for other pre-selections for example. I use this at different views:

https://github.com/minvws/nl-kat-coordination/blob/269c3b5bb992ce2b514d4d264585c0fe08ccfba2/rocky/reports/views/aggregate_report.py#L60-L63

https://github.com/minvws/nl-kat-coordination/blob/269c3b5bb992ce2b514d4d264585c0fe08ccfba2/rocky/reports/views/generate_report.py#L58-L61

https://github.com/minvws/nl-kat-coordination/blob/269c3b5bb992ce2b514d4d264585c0fe08ccfba2/rocky/reports/views/multi_report.py#L53-L56

https://github.com/minvws/nl-kat-coordination/blob/269c3b5bb992ce2b514d4d264585c0fe08ccfba2/rocky/reports/views/aggregate_report.py#L28